src/OrchardCore.Modules/OrchardCore.Media/Services/MediaFileStoreResolverMiddleware.cs

```diff
@@ -50,7 +51,7 @@ public async Task Invoke(HttpContext context)
         }
 
         var validatePath = context.Request.Path.StartsWithNormalizedSegments(_assetsRequestPath, StringComparison.OrdinalIgnoreCase, out var subPath);
-        if (!validatePath)
+        if (!validatePath || string.IsNullOrEmpty(_mediaFileStore.NormalizePath(subPath)))
```

Comment:
I think this makes sense, but the debug message below now doesn't match. Do we actually need that anyway?

/cc @MikeAlhayek

Reply:
Agreed, I would remove the message.

Reply:
Let us do it then

Reply:
Removed.